crates/evm/src/execute.rs

```diff
@@ -60,7 +60,7 @@ pub trait Executor<DB> {
         state: F,
     ) -> Result<Self::Output, Self::Error>
     where
-        F: FnMut(&State<DB>);
+        F: FnMut(&State<StateProviderDatabase<DB>>);
```

Comment:
what I wanted to do for this for a while now is introduce a trait for `State`, which would make this a lot more ergonomic. so for now I'm fine with that